### PR TITLE
Add developer mode health check for android device

### PR DESCRIPTION
**Pull request by keyonghan**
This add Android device developer mode health check: make sure it is on before reserving tests.

Related: https://github.com/flutter/flutter/issues/76165

### CODE REVIEW

File: device_doctor/lib/src/android_device.dart [hunk 1]
```diff
@@ -164,6 +165,24 @@ class AndroidDeviceDiscovery implements DeviceDiscovery {
     }
     return healthCheckResult;
   }
+
+  @visibleForTesting
```
Comment (godofredoc) [inline device_doctor/lib/src/android_device.dart:169]: Can we add some docs on what is this checking?

File: device_doctor/lib/src/android_device.dart [hunk 2]
```diff
@@ -164,6 +165,24 @@ class AndroidDeviceDiscovery implements DeviceDiscovery {
...
+  Future<HealthCheckResult> developerModeCheck({ProcessManager processManager}) async {
+    HealthCheckResult healthCheckResult;
+    try {
+      final String result = await eval(
+          'adb', <String>['shell', 'settings', 'get', 'global', 'development_settings_enabled'],
+          processManager: processManager);
+      if (result == '1') {
```
Comment (godofredoc) [inline device_doctor/lib/src/android_device.dart:176]: Can you please add a comment describing this is the actual output command not the exit code and 1 means enabled?

**Review by godofredoc — APPROVED** [review]
LGTM after fixing some nits

File: device_doctor/lib/src/android_device.dart (hunk 1 above)
Comment (keyonghan) [reply device_doctor/lib/src/android_device.dart:169]: Added.

File: device_doctor/lib/src/android_device.dart (hunk 2 above)
Comment (keyonghan) [reply device_doctor/lib/src/android_device.dart:180]: Added.

**Pull request merged**